src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -380,39 +353,47 @@ def _get_uniquely_named_nec_pnads(
     def get_name(cls) -> str:
         return "backchaining"
 
-    @staticmethod
-    def _specialize_pnad(
-        necessary_add_effects: Set[GroundAtom],
-        pnad: PartialNSRTAndDatastore,
-        ground_op: GroundNSRTOrSTRIPSOperator,
-    ) -> PartialNSRTAndDatastore:
-        """Given a PNAD, some necessary add effects and a grounding of the PNAD
-        such that the PNAD's add effects are a subset of the necessary add
-        effects, make the PNAD's add effects more specific ("specialize") so
-        that they cover these necessary add effects.
+    def _spawn_new_pnad(self, pnad: PartialNSRTAndDatastore,
+                        segment: Segment) -> PartialNSRTAndDatastore:
+        """Given a general PNAD and some segment with necessary add effects
+        that the PNAD must achieve, create a new PNAD ("spawn" from the most
+        general one) so that it has these necessary add effects.
```

Comment:
"these necessary add effects" -> "the necessary add effects contained in the given segment"